src/proto/proto_mesg_factory.cpp

```diff
@@ -77,23 +130,29 @@ class messaging_client : public grpc_client< Messaging >, public std::enable_sha
                                                         io_blob_list_t const& cli_buf) {
         return _generic_stub
             ->call_unary(cli_buf, request_name, NURAFT_MESG_CONFIG(mesg_factory_config->data_request_deadline_secs))
-            .deferValue([](auto&& response) -> NullResult {
-                if (response.hasError()) {
-                    LOGE("Failed to send data_service_request, error: {}", response.error().error_message());
-                    return folly::makeUnexpected(nuraft::cmd_result_code::CANCELLED);
-                }
-                return folly::unit;
-            });
+            .deferValue(
+                [weak_this = std::weak_ptr< messaging_client >(shared_from_this())](auto&& response) -> NullResult {
```

Comment:
can we catch `mc->_addr` as this is the only used field?

Reply:
I thought about that, but addr is an std::string which will cause.2 additional mallocs one for std::string copy (which might get inlined) and other for std::function itself since most compilers allow 24 bytes of inline and rest will do malloc for lamda capture. std::string is 32 bytes and weak_ptr is 16 bytes. So wanted to optimize for 99.999% code path.